pkgs/development/tools/mongosh/default.nix

```diff
@@ -1,15 +1,18 @@
-{ pkgs, stdenv, lib, testers, mongosh }:
+{ pkgs, lib, mongosh, buildNpmPackage, fetchzip }:
 
-let
-  nodePackages = import ./composition.nix {
-    inherit pkgs;
-    inherit (stdenv.hostPlatform) system;
-  };
-in
-nodePackages.mongosh.override {
-  passthru.tests.version = testers.testVersion {
-    package = mongosh;
+buildNpmPackage rec {
+  pname = "mongosh";
+  version = "1.7.1";
+  src = fetchzip {
```

Comment:
Upstream does provide a lock file, please use `fetchFromGitHub`: https://github.com/mongodb-js/mongosh

Reply:
This lockfile is not for mongosh package, it contains all dependencies used in the whole workspace, which contains lots of unused packages.

Reply:
So roughly how many unused packages (measured in disk space)? Having to manually generate a lock file would greatly increase the burden of maintenance.